astropy/convolution/convolve.py

```diff
@@ -811,15 +811,15 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
 
     if not np.all(newshape == arrayshape):
         if np.isfinite(fill_value):
-            bigarray = np.ones(newshape, dtype=complex_dtype) * fill_value
+            bigarray = npa.ones(newshape, dtype=complex_dtype) * fill_value
         else:
-            bigarray = np.zeros(newshape, dtype=complex_dtype)
+            bigarray = npa.zeros(newshape, dtype=complex_dtype)
         bigarray[arrayslices] = array
     else:
         bigarray = array
 
     if not np.all(newshape == kernshape):
-        bigkernel = np.zeros(newshape, dtype=complex_dtype)
+        bigkernel = npa.zeros(newshape, dtype=complex_dtype)
         bigkernel[kernslices] = normalized_kernel
     else:
         bigkernel = normalized_kernel
```

Comment:
```suggestion
        bigkernel = normalized_kernel

    if hasattr(bigarray, 'chunks'):
        bigarray = bigarray.rechunk(tuple((dim,) for dim in bigarray.shape))
    if hasattr(bigkernel, 'chunks'):
        bigkernel = bigkernel.rechunk(tuple((dim,) for dim in bigkernel.shape))
```
else this is failing on larger arrays with
```python
ValueError: Dask array only supports taking an FFT along an axis that 
has a single chunk. An FFT operation was tried on axis 0 
which has chunks (2420, 2420). To change the array's chunks use dask.Array.rechunk.
```
The same needs to be done on the other arrays `fftn` or `ifftn` is called on, so it would perhaps be better to define two wrappers that directly operate on `fftn(bigarray.rechunk(...))`.

Reply:
yikes, that's probably a complete blocker to this approach

Reply:
Don't see such a big blocker here; couldn't you still take the user-input function as
```python
def _fftn(arr):
    if hasattr(arr, 'chunks'):
        return fftn(arr.rechunk(tuple((dim,) for dim in arr.shape)))
    else:
        return fftn(arr)
```
(don't know of course what the performance implications are when getting to really big arrays, but that seems a fundamental limit for Dask+FFT)?

Reply:
right, sure, that's not crazy.  But there are so many things one has to do to get a dask array to work with convolution, we're almost at the point of writing a separate function.

Reply:
Yes, this one is very `ndarray`-tuned.
The main show-stoppers to me seemed the masked and `nan_treatment='fill'` cases, but I think support for those could well be postponed for a later implementation, or accept that they will trigger load/compute operations for the time being (I have now idea even about the support for masks in Dask).